packages/special-pages/pages/new-tab/app/privacy-stats/PrivacyStats.js

```diff
@@ -0,0 +1,226 @@
+            {none && (
+                <p className={styles.title}>{t('trackerStatsNoRecent')}</p>
+            )}
+            {some && (
+                <p className={styles.title}>{alltimeTitle}</p>
+            )}
+            <span className={styles.expander}>
+                <button
+                    type="button"
+                    className={styles.toggle}
+                    onClick={onToggle}
+                    aria-pressed={pressed}
+                    aria-label={t('trackerStatsToggleLabel')}
+                    hidden={trackerCompanies.length === 0}
+                >
+                    <Chevron/>
+                </button>
+            </span>
+            <p className={styles.subtitle}>
+                {recent === 0 && t('trackerStatsNoActivity')}
+                {recent > 0 && recentTitle}
+            </p>
+        </div>
+    )
+}
+
+/**
+ * @param {object} props
+ * @param {TrackerCompany[]} props.trackerCompanies
+ * @param {import("./PrivacyStatsProvider.js").Ui} [props.ui]
+ * @param {(evt: import("./PrivacyStatsProvider.js").Events) => void} [props.send]
+ */
+export function Body ({ trackerCompanies, send, ui }) {
+    const max = trackerCompanies[0]?.count ?? 0
+    const [formatter] = useState(() => new Intl.NumberFormat())
+
+    const exited = useCallback((e) => {
+        if (e.target.classList.contains(styles.exiting)) {
+            send?.({ kind: 'did-exit' })
+        }
+        if (e.target.classList.contains(styles.entering)) {
+            send?.({ kind: 'did-enter' })
+        }
+    }, [send])
+
+    const bodyClasses = cn({
+        [styles.list]: true,
+        [styles.exiting]: ui === 'exiting',
+        [styles.entering]: ui === 'entering',
+        [styles.entered]: ui === 'entered'
+    })
+    return (
+        <ul className={bodyClasses} onAnimationEnd={exited} onTransitionEnd={exited}>
+            {trackerCompanies.map(company => {
+                const percentage = Math.min((company.count * 100) / max, 100)
+                const valueOrMin = Math.max(percentage, 10)
+                const inlineStyles = {
+                    width: `${valueOrMin}%`
+                }
+                const countText = formatter.format(company.count)
+                return (
+                    <li key={company.displayName}>
+                        <div className={styles.row}>
+                            <div className={styles.company}>
+                                <CompanyIcon company={company}/>
+                                <span className={styles.name}>{company.displayName}</span>
+                            </div>
+                            <span className={styles.count}>{countText}</span>
+                            <span className={styles.bar}></span>
+                            <span className={styles.fill} style={inlineStyles}></span>
+                        </div>
+                    </li>
+                )
+            })}
+        </ul>
+    )
+}
+
```

Comment:
Can we have comments above PrivacyStatsCustomized and PrivacyStatsConsumer explaining their purpose? Might provide context for people looking at this code for the first time, and also to solidify the Customized/Consumer naming pattern, which I quite like.